src/main/java/de/tum/in/www1/artemis/service/connectors/AbstractVersionControlService.java

```diff
@@ -173,4 +173,20 @@ public String getOrRetrieveBranchOfExercise(ProgrammingExercise programmingExerc
     public String getDefaultBranchOfArtemis() {
         return defaultBranch;
     }
+
+    /**
+     * Determines if a user should get read-only or also write permissions for their repository.
+     *
+     * @param programmingExercise The programming exercise the repository belongs to.
+     * @return The permissions the user should receive for a repository.
+     */
+    protected RepositoryPermissions determineRepositoryPermissions(final ProgrammingExercise programmingExercise) {
+        // NOTE: null values are interpreted as offline IDE is allowed
+        if (Boolean.FALSE.equals(programmingExercise.isAllowOfflineIde())) {
```

Comment:
Should we move this check into isAllowOfflineIde, since no occurrence actually needs the null value and all calls to this method always have to wrap the call into an equals check

Reply:
Probably a good idea. There are a few more `Boolean` attributes in the programming exercise, though, that should be changed as well in the same refactoring (online editor enabled, static code analysis enabled, publish build plan, …).
They are all used in quite a few places, so any change there should be part of a separate refactoring PR to ensure there that the logic does not change. ⇒ opened as issue #6079